go.mod

```diff
@@ -1,5 +1,7 @@
 module dario.cat/mergo
 
-go 1.13
+go 1.21.0
```

Comment:
I see why the bump, but maybe this means we just need to use something that is not YAML for testing. I appreciate the PR, but I think it's better to remove the dependency instead.

Reply:
Update: we have two options, migrate from YAML testdata files to JSON or write a toy YAML parser just for testing purpose. After looking at the testdata files, I'm leaning towards migrating to JSON.